File: DesignPatternsExercises/java/MementoDesignPattern/Exercise/Warehouse.java
import java.util.ArrayList;
import java.util.List;

public class Warehouse {
	private List<__________(5)__________> homeTheatreList = new ArrayList<>();

	public void addMemento(Memento memento) {
		 homeTheatreList.add(memento);
	}

	public Memento __________(6)__________(int index) {
		return homeTheatreList.get(index);
	}
}

File: DesignPatternsExercises/ruby/memento_design_pattern/solution/warehouse.rb
class Warehouse
  def initialize
    # __________(9)__________ = []
    @home_theatre_list = []
  end

  # def __________(10)__________(memento)
  def add_memento(memento)
    @home_theatre_list << memento
  end

  def memento(index)
    @home_theatre_list[index]
  end
end
